pkgs/top-level/all-packages.nix

```diff
@@ -20251,6 +20251,42 @@ with pkgs;
 
   nixops-dns = callPackage ../tools/package-management/nixops/nixops-dns.nix { };
 
+  /*
+   * Evaluate a NixOS configuration using this evaluation of NixPkgs.
+   *
+   * With this function you can write, for example, a package that
+   * depends on a custom virtual machine image.
+   *
+   * Parameter: A module, path or list of those that represent the
+   *            configuration of the NixOS system to be constructed.
+   *
+   * Result:    An attribute set containing packages produced by this
+   *            evaluation of NixOS, such as toplevel, kernel and
+   *            initialRamdisk.
+   *            The result can be extended in the modules by defining
+   *            extra options in system.build.
+   *
+   * For more information about writing NixOS modules, see
+   * https://nixos.org/nixos/manual/index.html#sec-writing-modules
+   *
+   * If you want to call this function from a system that is not
+   * supported by NixOS, you will need to initialise NixPkgs with a
+   * custom system attribute.
+   */
+  nixos = configuration:
+    (import (pkgs.path + "/nixos/lib/eval-config.nix") {
+      inherit (pkgs) system;
+      modules = [(
+                  {lib,...}: {
+                    config.nixpkgs.pkgs = lib.mkDefault self;
```

Comment:
nit: `config.nixpkgs.pkgs = lib.mkDefault pkgs;`  Otherwise this would only capture packages from all-packages and none of the one added by any of the overlays. This is a defect of the `all-packages.nix` file.